framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -529,8 +529,7 @@ protected void addInformationFromPreconditions(
                                 false);
                 info.insertValue(expr, annotation);
             } catch (FlowExpressionParseException e) {
-                // report errors here
-                analysis.checker.report(e.getResult(), methodTree);
```

Comment:
I think it's a neutral thing to move the checks (though moving the checks means that the expressions will be parsed twice, which isn't a good thing). The only type systems that have postcondition annotations are those that use flow refinement, so it doesn't benefit more type systems to move these errors.  

I agree that flow refinement should not issue type-checking errors, but flow refinement might issue other errors/warnings related to flow refinement.  I would argue that these errors are related to flow refinement rather than type-checking errors. So, I don't think it is a problem to leave the checks as is.  However, if you feel strongly that they should be in the BaseTypeVisitor, then it is ok with me if you move them.